languages/tree-sitter-stack-graphs-javascript/src/stack-graphs.tsg

```diff
@@ -3476,6 +3476,287 @@ inherit .return_or_yield
   edge @class.after_scope -> @name.after_scope
 }
 
+(jsx_element
+  open_tag:(_)@open_tag
+  close_tag:(_)@close_tag)@jsx_element {
+
+  node @jsx_element.before_scope
+  node @jsx_element.after_scope
+  node @jsx_element.value
+
+  edge @open_tag.before_scope -> @jsx_element.before_scope
+  edge @jsx_element.after_scope -> @close_tag.after_scope
+
+  ; LATER-TODO scope propagation through empty JSX children
+  ; currently unsupported by tree sitter queries
+  ; THIS IS A HUGE HACK AND MUST BE FIXED
+  edge @close_tag.before_scope -> @open_tag.after_scope
```

Comment:
With TSG we can actually do this. See https://github.com/github/stack-graphs/blob/main/languages/tree-sitter-stack-graphs-javascript/src/stack-graphs.tsg#L1637-L1641 for example.

Reply:
Done!